domain/consensus/processes/coinbasemanager/coinbasemanager.go

```diff
@@ -182,97 +177,44 @@ func acceptanceDataFromArrayToMap(acceptanceData externalapi.AcceptanceData) map
 // should have. This is mainly used for determining how much the coinbase for
 // newly generated blocks awards as well as validating the coinbase for blocks
 // has the expected value.
-//
-// Further details: https://hashdag.medium.com/kaspa-launch-plan-9a63f4d754a6
-func (c *coinbaseManager) CalcBlockSubsidy(blockHash *externalapi.DomainHash) (uint64, error) {
+func (c *coinbaseManager) CalcBlockSubsidy(stagingArea *model.StagingArea, blockHash *externalapi.DomainHash) (uint64, error) {
 	if blockHash.Equal(c.genesisHash) {
 		return c.subsidyGenesisReward, nil
 	}
-
-	return c.maxSubsidy, nil
-}
-
-func (c *coinbaseManager) calculateAveragePastSubsidy(stagingArea *model.StagingArea, blockHash *externalapi.DomainHash) (*big.Rat, error) {
-	const subsidyPastWindowSize = 100
-	blockWindow, err := c.dagTraversalManager.BlockWindow(stagingArea, blockHash, subsidyPastWindowSize)
+	blockDaaScore, err := c.daaBlocksStore.DAAScore(c.databaseContext, stagingArea, blockHash)
 	if err != nil {
-		return nil, err
-	}
-	if len(blockWindow) == 0 {
-		return new(big.Rat).SetFrac64(int64(c.subsidyGenesisReward), 1), nil
+		return 0, err
 	}
-
-	pastBlocks, err := c.blockStore.Blocks(c.databaseContext, stagingArea, blockWindow)
-	if err != nil {
-		return nil, err
+	if blockDaaScore < c.deflationaryPhaseDaaScore {
+		return c.preDeflationaryPhaseBaseSubsidy, nil
 	}
 
-	pastBlockSubsidySum := int64(0)
-	for _, pastBlock := range pastBlocks {
-		coinbaseTransaction := pastBlock.Transactions[transactionhelper.CoinbaseTransactionIndex]
-		_, _, pastBlockSubsidy, err := c.ExtractCoinbaseDataBlueScoreAndSubsidy(coinbaseTransaction)
-		if err != nil {
-			return nil, err
-		}
-		pastBlockSubsidySum += int64(pastBlockSubsidy)
-	}
-	return big.NewRat(pastBlockSubsidySum, int64(len(blockWindow))), nil
+	blockSubsidy := c.calcDeflationaryPeriodBlockSubsidy(blockDaaScore)
+	return blockSubsidy, nil
 }
 
-func (c *coinbaseManager) calculateMergeSetSubsidySum(stagingArea *model.StagingArea, blockHash *externalapi.DomainHash) (*big.Rat, error) {
-	ghostdagData, err := c.ghostdagDataStore.Get(c.databaseContext, stagingArea, blockHash, true)
-	if !database.IsNotFoundError(err) && err != nil {
-		return nil, err
-	}
+func (c *coinbaseManager) calcDeflationaryPeriodBlockSubsidy(blockDaaScore uint64) uint64 {
+	// We define a year as 365.25 days and a month as 365.25 / 12 = 30.4375
+	// secondsPerMonth = 30.4375 * 24 * 60 * 60
+	const secondsPerMonth = 2629800
+	monthsSinceDeflationaryPhaseStarted := (blockDaaScore - c.deflationaryPhaseDaaScore) / secondsPerMonth
 
-	// If there's ghostdag data with trusted data we prefer it because we need the original merge set non-pruned merge set.
-	if database.IsNotFoundError(err) {
-		ghostdagData, err = c.ghostdagDataStore.Get(c.databaseContext, stagingArea, blockHash, false)
-		if err != nil {
-			return nil, err
-		}
-	}
+	// enlargedDeflationConstant is the estimated deflation constant `2^1/12` multiplied by a big
```

Comment:
Can we add the following full explanation?:

/*
Denote `B`, `M` to be `deflationaryPhaseBaseSubsidy` and `monthsSinceDeflationaryPhaseStarted` respectively.

We want to compute the following function in a deterministic way:

`subsidy = B/(2^(1/12))^M`

Set `C = enlargedDeflationConstant` and `F = enlargeFactor` large enough s.t. C/F ~= 2^(1/12).

We start with `c=1 (enlargedDeflationMultiplier)`, `f=1 (enlargeFactorDivisor)` and repeat c *= C, f *= F for M times
 s.t. eventually  `c/f ~= (2^(1/12))^M`.  

Finally we compute `subsidy = B * (f/c)`

*/ 

P.S. 1: I would love it if var names were actually short like in this snippet^, it's much harder to read equations with sentences then with simple in-context clear names such as `c`, `f`

P.S. 2: with current base subsidy, the subsidy will reach zero when `M=425`, so effectively we can pre-compute this and place it in a table. Just saying. (for instance benchmarking this function shows that with `M=30*12` it takes ~2ms)